odl/test/solvers/functional/default_functionals_test.py

```diff
@@ -516,5 +516,61 @@ def test_moreau_envelope_l2_sq(space, sigma):
                             x * 2 / (1 + 2 * sigma))
 
 
+def test_bregman_functional_no_gradient(space):
+    """Test that the Bregman distance functional fails if the underlying
+    functional does not have a gradient and no subgradient operator is
+    given."""
+
+    ind_func = odl.solvers.IndicatorNonnegativity(space)
+    point = noise_element(space)
+
+    # Indicator function has no gradient, hence one cannot create a bregman
+    # distance functional
+    with pytest.raises(NotImplementedError):
+        odl.solvers.BregmanDistance(ind_func, point)
+
+    # If a subgradient operator is given separately, it is possible to create
+    # an instance of the functional
+    subgrad_op = odl.IdentityOperator(space)
+    bregman_dist = odl.solvers.BregmanDistance(ind_func, point, subgrad_op)
+
+    # In this case we should be able to call the gradient of the bregman
+    # distance, which would give us a subgradient
+    x = np.abs(noise_element(space))
+    expected_result = subgrad_op(x) - subgrad_op(point)
+    assert all_almost_equal(bregman_dist.gradient(x), expected_result)
+
+
+def test_bregman_functional_l2_squared(space, sigma):
+    """Test for the Bregman distance functional, using l2 norm squared as
+    underlying functional."""
+    sigma = float(sigma)
+
+    l2_sq = odl.solvers.L2NormSquared(space)
+    point = noise_element(space)
+    subgrad_op = odl.ScalingOperator(space, 2.0)
```

Comment:
Why does this need to be provided?

Reply:
It does not need to be provided. It is a relic since I wrote this test before the other test, and wanted to check that it in fact return the expected gradient also when a subdifferential operator is given. But given the test above that now tests this, I will remove it.